### PR TITLE
Dashboard admin speaker

### PR DESCRIPTION
**Pull request by PurpurTentakel97**
this PR provides speaker in the admin event dashboard to enter the `visible_from` field.

### CODE REVIEW

File: src/routes/dashboard/admin/events/+page.ts [hunk 1]
```diff
@@ -17,5 +17,6 @@ export async function load({ fetch }: { fetch: typeof globalThis.fetch }): Promi
 
     return {
         allEvents,
+        allSpeaker: [] // will be filled inside the componend
```
Comment (mgerhold) [inline src/routes/dashboard/admin/events/+page.ts:20]:
```suggestion
        allSpeaker: [] // will be filled inside the component
```

File: src/routes/dashboard/admin/events/eventsHelper.ts [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+
+function trimOrNull(entry: string): string | null {
+    if (entry.trim().length === 0) {
+        return null;
+    }
+    return entry.trim();
+}
+function trim(entry: string): string {
+    return entry.trim();
+}
```
Comment (mgerhold) [inline src/routes/dashboard/admin/events/eventsHelper.ts:18]: Is this function really used somewhere? And if so: Why? This seems unnecessary.

File: src/types/saveMessageType.ts [hunk 3]
```diff
@@ -7,3 +7,19 @@ export enum SaveMessageType {
...
+        return lhs === reference || rhs === reference;
+    }
+
+    if (check(lhs, rhs, SaveMessageType.Error)) {
+        return SaveMessageType.Error;
+    }
+
+    if (check(lhs, rhs, SaveMessageType.None)) {
+        return SaveMessageType.None;
+    }
```
Comment (mgerhold) [inline src/types/saveMessageType.ts:22]: This code looks like "I know lambdas, so I have to use lambdas"! 😅 Wouldn't the following code not only be shorter, but also easier to understand?
```suggestion
    if (lhs === SaveMessageType.Error || rhs === SaveMessageType.Error) {
        return SaveMessageType.Error;
    }

    if (lhs === SaveMessageType.None || rhs === SaveMessageType.None) {
        return SaveMessageType.None;
    }
```

File: src/routes/dashboard/admin/events/eventsHelper.ts [hunk 4]
```diff
@@ -52,7 +54,32 @@ export function convertSaveData(data: SetAdminEvent): SetAdminEvent {
...
+
+export async function loadSpeaker(fetch: Function, eventId: number): Promise<DashboardAllEventSpeaker> {
+    const allSpeakerPromise = fetch(apiUrl(`/api/dashboard/admin/event/${eventId}/speaker`));
+
+    const allSpeaker = await checkAndParseInputDataAsync<DashboardAllEventSpeaker>(
+        await allSpeakerPromise,
+        dashboardAllEventSpeakerScheme,
+        `Serveranfrage für alle Speaker im event ${eventId} nicht erfolgreich. throw error(406)`,
+        `Unerwartete Daten für alle Speker im Event ${eventId}. throw error(406)`
```
Comment (peddermaster2) [inline src/routes/dashboard/admin/events/eventsHelper.ts:76]:
```suggestion
        `Unerwartete Daten für alle Speaker im Event ${eventId}. throw error(406)`
```
Oder besser noch "Vortragende"?

**Review by mgerhold — APPROVED** [review]

**Pull request merged**